awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -410,6 +410,47 @@ def modify_associations(self, association_endpoint, new_association_list):
             else:
                 self.fail_json(msg="Failed to associate item {0}".format(response['json'].get('detail', response['json'])))
 
+    def copy_item(self, existing_item, copy_from_name_or_id, new_item_name, endpoint=None, item_type='unknown', copy_lookup_data={}):
+
+        if existing_item is not None:
```

Comment:
One other o'dark 30 thought I had was that this check makes your playbooks not idempotent.... i.e. if you had a playbook with a copy_from you could run it once but the second run causes a failure in the module halting your playbook. You could get around this by making your task with the copy_from wrapped in a when clause but then you have to do the lookup yourself out side. I am questioning if we want this to be a failure or if want an existing item to just nullify the copy from? i.e. if the target object already exists we just do a return here and the copy never takes place but the rest of the module runs. We could document this behavior and then a playbook can be run over and over w/o a when condition on the task. Thoughts?

Reply:
Definitely doable, as a return would just return back to the module to continue as normal, IF we do this, We should make changed = false, with the rest of the module determining if it then gets changed to true, and a "copied" = false if it doesn't, and true if it does. That way the user can parse the results easier. Additionally maybe change this line to self.warn?

Reply:
Made said changes and updated test logic.